packages/shorebird_cli/lib/src/validators/shorebird_yaml_validator.dart

```diff
@@ -0,0 +1,28 @@
+import 'package:shorebird_cli/src/shorebird_process.dart';
+import 'package:shorebird_cli/src/validators/validators.dart';
+
+class ShorebirdYamlValidator extends Validator {
+  ShorebirdYamlValidator({required this.hasShorebirdYaml});
+  // Adding as Function
+  /// Returns true if the project has a shorebird.yaml file.
+  final bool Function() hasShorebirdYaml;
```

Comment:
Why are we injecting a function to do the validation? Thoughts on performing the validation directly in the validator? We can still unit test it and it adds one less level of indirection imo. Let me know what you think!

Reply:
Yup, make sense. I think we should do it directly inside validator. I'll update